scripts/bootloader/keygen.py

```diff
@@ -11,6 +11,22 @@
 import sys
 
 
+def gen_key():
```

Comment:
For clarity, it is best to avoid clipping words that don't need it. Perhaps we could rename this to something like `generate_legal_key`? It would go nicely with a docstring comment explaining its purpose and why the legal check is important.